### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by JoshuaPK**
This is essentially the same changes I made to the previous codebase (when the backend and frontend were separate).

### CODE REVIEW

File: paperwork-backend/paperwork_backend/docsearch.py [hunk 1]
```diff
@@ -264,16 +264,20 @@ def __init__(self, rootdir, indexdir=None, language=None,
...
-
-        indexdir = os.path.join(indexdir, "index")
+
+        if index_in_workdir:
+            base_data_dir = self.fs.unsafe(rootdir)
+            localdir = base_data_dir
+            indexdir = os.path.join(base_data_dir, "index")
+
+        elif indexdir is None:
+            base_data_dir = os.getenv(
```
Comment (jflesch) [inline paperwork-backend/paperwork_backend/docsearch.py]: if `not index_in_workdir` and `indexdir is not None` then `base_data_dir` won't be set, but still be passed to `self.index.open()`.

Comment (JoshuaPK) [reply paperwork-backend/paperwork_backend/docsearch.py]: I corrected this by moving the base_data_dir = self.fs.unsafe(rootdir) to before the "if index_in_workdir" statement.

**Comment by jflesch** [comment]
Almost there. There are just 2 minor PEP8 issues remaining:

https://origami.openpaper.work/#/builders/2/builds/264/steps/2/logs/stdio

```
paperwork_backend/config.py:143:80: E501 line too long (82 > 79 characters)
paperwork_backend/shell.py:55:80: E501 line too long (129 > 79 characters)
```

(you can use `make check` (requires `flake8`) to check your changes)

**Comment by JoshuaPK** [comment]
When I ran `make check` I got the following.  You didn't mention these- are they inconsequential, or are our environments slightly different?

`paperwork_backend/deps.py:8:1: E722 do not use bare except'
paperwork_backend/shell_cmd.py:17:1: E722 do not use bare except'
paperwork_backend/shell_cmd.py:22:1: E722 do not use bare except'
paperwork_backend/util.py:33:1: E722 do not use bare except'
paperwork_backend/util.py:44:1: E722 do not use bare except'
paperwork_backend/shell.py:56:17: E128 continuation line under-indented for visual indent
paperwork_backend/common/doc.py:31:1: E722 do not use bare except'
make[1]: *** [check] Error 1
make[1]: Leaving directory /home/josh/dev/Projects/DocMgmt/openpaperwork/oxygen-workspace/OpenPaperwork/src/paperwork/paperwork-backend'
make: *** [paperwork-backend_check] Error 2`

**Comment by jflesch** [comment]
Interesting, I don't get those warnings on Debian stable (see the buildbot output) nor Ubuntu 17.10. I'll fix them later.

**Comment by jflesch** [comment]
https://origami.openpaper.work/#/builders/2/builds/265
Close enough. I'll fix that last one later. Let's merge :)

**Pull request merged**